### PR TITLE
Add ScopedMauiContext Wrapper

### LINKED ISSUE #2353: [Bug] Nesting NavigationPage inside of TabbedPage breaks page layout
### Description

Using build `6.0.100-rc.2.6746`

If you nest a NavigationPage inside of a TabbedPage as a child, the contents of the tab are rendered with a height and width of zero. This appears to only happen on Android. 

https://user-images.githubusercontent.com/898335/131753977-df7a179c-2cfb-4b7b-a072-baeced0dcddf.mp4

### Steps to Reproduce

Run this sample project.
[MauiTabTest.zip](https://github.com/dotnet/maui/files/7094716/MauiTabTest.zip)

### Expected Behavior

The pages would render correctly.

### Actual Behavior

They are broken.

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

When nesting fragments android requires you to pass down the relevant LayoutInflater/ChildFragmentManager.

Currently the only mechanism for passing this data in is IScopedMauiContext so this PR adds a set of properties to IScopedMauiContext that allows you to create a Context/FragementManager/LayoutInflater that's scoped to a set of child views. 

This will also be helpful for cases where we need to use a ContextThemeWrapper.

- Fixes #2353

### Additions made ###
- All features added are internal. Once we figure out exactly how we want to do Scoped Maui Context feature we will have public APIs

### Things that this PR doesn't fix
- When you go to the TabbedPage that I added you'll notice things look a little off and navigation doesn't quite work on the internal pages. Those will be fixed in later PRs. This PR makes it so the app doesn't crash and something shows up at all.

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

### CODE REVIEW

**Review by Pakon2543 — APPROVED** [review]

File: src/Compatibility/Core/src/Android/AppCompat/Platform.cs [hunk 1]
```diff
@@ -359,17 +368,12 @@ internal static IVisualElementRenderer CreateRenderer(VisualElement element, Con
 			}
 
 			renderer.SetElement(element);
-			return renderer;
-		}
-
-		internal static IVisualElementRenderer CreateRenderer(VisualElement element, AndroidX.Fragment.App.FragmentManager fragmentManager, Context context)
```
Comment (PureWeen) [inline src/Compatibility/Core/src/Android/AppCompat/Platform.cs:365]: nothing exciting here. Only one API in Forms was using this so I just moved it all into the same `CreateRenderer `method

File: src/Compatibility/Core/src/Android/VisualElementTracker.cs [hunk 2]
```diff
@@ -96,7 +96,7 @@ public void UpdateLayout()
 				formsViewGroup.MeasureAndLayout(MeasureSpecFactory.MakeMeasureSpec(width, MeasureSpecMode.Exactly), MeasureSpecFactory.MakeMeasureSpec(height, MeasureSpecMode.Exactly), x, y, x + width, y + height);
 				Performance.Stop(reference, "MeasureAndLayout");
 			}
-			else if ((aview is LayoutViewGroup || aview is PageViewGroup) && width == 0 && height == 0)
+			else if ((aview is LayoutViewGroup || aview is PageViewGroup || aview is NavigationLayout) && width == 0 && height == 0)
```
Comment (PureWeen) [inline src/Compatibility/Core/src/Android/VisualElementTracker.cs:99]: Yay VisualElementTracker!

File: src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs [hunk 3]
```diff
@@ -2,10 +2,10 @@
 using Maui.Controls.Sample.Pages.Base;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.Graphics;
-using Microsoft.AspNetCore.Components.Web;
-using Maui.Controls.Sample.Pages.Others;
 
 #if NET6_0_OR_GREATER
+using Microsoft.AspNetCore.Components.Web;
```
Comment (PureWeen) [inline src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs:7]: This fixes our classic solutions

**Review by Pakon2543 — APPROVED** [review]

File: src/Controls/src/Core/Platform/Android/ControlsNavigationManager.cs [hunk 4]
```diff
@@ -156,11 +156,6 @@ protected override void OnDestinationChanged(NavController navController, NavDes
 			{
 				ToolbarTracker.Target = CurrentPage;
 			}
-
```
Comment (PureWeen) [inline src/Controls/src/Core/Platform/Android/ControlsNavigationManager.cs:159]: Moved this up to core

File: src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs (hunk 3 above)
Comment (Eilon) [reply src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs:9]: Hmm but this same namespace is also imported a few lines up? How would this be needed?

Comment (Eilon) [reply src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs:8]: Ah, that makes sense now (you removed from the non-`#if`'ed part. That looks good then!

Comment (PureWeen) [reply src/Controls/samples/Controls.Sample/Pages/Others/BlazorPage.xaml.cs:7]: yea I think a rebase or something added them back in :-/ I meant to delete those

File: src/Core/src/Platform/MauiContext.Android.cs [hunk 5]
```diff
@@ -13,7 +16,7 @@ public MauiContext(IServiceProvider services, Context context)
 			_context = new WeakReference<Context>(context ?? throw new ArgumentNullException(nameof(context)));
 		}
 
-		public MauiContext(Context context)
+		internal MauiContext(Context context)
```
Comment (rmarinho) [inline src/Core/src/Platform/MauiContext.Android.cs:19]: So now users can't create a MauiContext ?! they have to create a ScopedMauiContext ?

**Review by rmarinho — APPROVED** [review]

**Pull request merged**